x/authz/expected_keepers.go

```diff
@@ -7,9 +7,9 @@ import (
 
 // AccountKeeper defines the expected account keeper (noalias)
 type AccountKeeper interface {
-	GetAccount(ctx sdk.Context, addr sdk.AccAddress) authtypes.AccountI
-	NewAccountWithAddress(ctx sdk.Context, addr sdk.AccAddress) authtypes.AccountI
-	SetAccount(ctx sdk.Context, acc authtypes.AccountI)
+	GetAccount(ctx sdk.Context, addr sdk.AccAddress) authtypes.IAccount
```

Comment:
> This is a huge API breaking change IMO, for all chains

Agreed.

> I also don't see a strong need to switch

Yea, I was conflicted as well so I made sure to double-check if I should proceed before making any changes: https://github.com/cosmos/cosmos-sdk/issues/13254#issuecomment-1245420068

>  (just because some guy wrote about it in a medium article?)

It seems to be the "Go" standard. I can find more references and added them to the PR. Just lmk if I should invest the time

> Add type aliases, e.g. AccountI -> IAccount, so that everything is backwards-compatible. Add deprecation godoc on all old interfaces.

Done. See #13279

Here's an example:

<img width="529" alt="Screen Shot 2022-09-14 at 11 21 46 AM" src="https://user-images.githubusercontent.com/1892194/190196117-cfa7f75f-db78-456f-9999-d3e09c62211f.png">

> Rip off the old interfaces, but probably not in 0.47

Could you leave a comment in #13279 on next steps and timelines?

Reply:
I appreciate your efforts here @Olshansk and I know we're not really treating the SDK as 1.0 yet so some breaking changes are okay. At the same time, I think we really need to measure the value of breaking changes like this. I generally think changing things just because some authority says the style should be different isn't a great reason to force downstream changes on clients... So my preference would generally for us not to be spending time on these sort of changes (and then forcing our users to spend time on them), if there isn't some larger benefit. `AccountI`, etc. may seem clunky but it also seems like not a big deal...

Reply:
Understood.

I confirmed the scope of work before starting in the original tickets, tended to the comments in the PR, documented the future work, tested the changes, and made it backwards compatible with the goal of better code health before the SDK reaches 1.0

If you still believe I should close out the PR and issue without a merge, please lmk.

Reply:
The aliases address backwards compatibility. Also if its an issue for downstream users, we don't _have_ to remove them for 1.0

@amaurym @marbar3778 what do you think?

Reply:
Sorry guys, I posted in the issue that I was Ok with this change but I didn't realize how much of a breaking change it would be. I personally don't think aliases are worth it if the existing type names exist. Perhaps we consider closing the PR? 

Appreciate the efforts @Olshansk 🙏

Reply:
Let me know what the team decides. 

I realize these are just tiny NITs that don't matter in the grand scheme of things, but the reason I personally drive for it is to:
1. Set culture for best coding practices moving forward
2. Avoid a _"broken window theory"_ theory when it comes to code quality
3. Reduce cognitive load for newcomers to the space
4. I've found that **implementing** the _"real interesting work"_ is often just a couple hundred lines of code, that's heavily dependant on the code (not the design)

Reply:
I think that @amaurym's suggestion is valid and fine to proceed with.

1. Add aliases
2. Mark legacy types deprecated
3. Remove legacy deprecated types in subsequent release.

Reply:
@alexanderbez see the screenshot I posted earlier in the thread where I created an alias for the FeeAllowance interface, along with an alias and a TODO on top.

Did you have something else in mind?

Reply:
@Olshansk I don't understand why the diff is so large? I thought we were just defining new types and creating aliases and adding deprecation comments to current types?

Reply:
**Orignail goal:** Replace all interfaces of the form `FooI` with `IFoo`throughout the codebase.

**Original reasoning**: 
- Promote Goland best practices throughout the codebase
- Promote devs to create future interfaces using best practices
- Promote future usage of previous interfaces using the new name

**Changes made**:
- Updated all usages
- Updated all documentation

**Problem**:
- Lack of backwards compatibility

Changes made:
- For each interface changed, we did the following:
```go
// TODO(#13279): Remove this alias in a future release
type TestI ITest

type ITest interface {
Original change: 
```

What this enables:
- Allow all the changes made to be kept in place
- Ticket #13279 tracks the long-term deprecation of the old interface
- When #13279 is tackled in the future, all the dev (myself or other) will need to do, is:
   1. Grep for `TODO(#13304)
   2. Delete the comment + the alias below

Reply:
So you mean you defined new proto types? Or just aliases? Idk...just seems like such a large diff and tech debt to maintain for a version just to change type names.

Reply:
I implemented the approach @amaurym suggested in the first message in this thread.

Given the long async back and forth, is there a cosmos contributor hour where we can discuss it over a call?

Reply:
@amaurym Could you take a look at the latest changes and see if it aligns with your suggestion?

Reply:
My personal take is that this PR sees a lot of resistance, and doesn't have a clear motivation from SDK users (i.e. no-one came to us about AccountI vs IAccount).

@Olshansk I really appreciate your work here, but my recommendation would be to simply to close the PR. I'm sorry about this, we should have better discussed this in the issue beforehand.

However, if you feel strongly about this going through, please join the SDK community call this Thursday 6-7pm CEST, and we can quickly discuss this with a wider audience before moving forward.

Reply:
Will do!